e2e/files.spec.ts

```diff
@@ -356,7 +356,10 @@ testAuth.describe('Files', () => {
       await expect(
         userPage.getByRole('heading', { name: userProject.name }),
       ).toBeVisible()
-      await userPage.getByRole('link', { name: 'Files', exact: true }).click()
+      await userPage
+        .getByRole('link', { name: 'Files', exact: true })
+        .first()
+        .click()
```

Comment:
E2E test changes are from #5497, which also had one of the React component fixes in the same commit, so it ended up also being pulled in here.